query/filter/logical.go

```diff
@@ -111,6 +100,57 @@ func (a *AndFilter) String() string {
 	return str + "}"
 }
 
+func (a *AndFilter) ToSearchFilter() []string {
+	var selectors []*Selector
+	var logical []Filter
+	for _, f := range a.filter {
+		if s, ok := f.(*Selector); ok {
+			selectors = append(selectors, s)
+		} else {
+			logical = append(logical, f)
+		}
+	}
+
+	var str string
+	for i, s := range selectors {
+		// first "&&" all selectors
+		if i != 0 {
+			str += "&&"
+		}
+		str += s.ToSearchFilter()[0]
+	}
+
+	var flattened []string
+	if len(logical) > 0 {
+		flattened = a.flattenAnd(str, logical)
+	}
+	if len(flattened) == 0 {
+		flattened = append(flattened, str)
+	}
+	return flattened
+}
+
+func (a *AndFilter) flattenAnd(soFar string, filters []Filter) []string {
+	var combs []string
+
+	for _, e := range filters[0].ToSearchFilter() {
+		var temp = soFar
+		if len(temp) > 0 {
```

Comment:
```
if len(soFar) > 0 {
  soFar += "&&"
}
for .... {
  if len(filters) > 1 {
    combs = append(combs, a.flattenAnd(soFar + e, filters[1:])...)
  } else {
    combs = append(combs, soFar+e)
  }
}
```

Reply:
that won't work if this for loop runs more than once i.e. if the filters[0].ToSearchFilter() returns a list of more than 1 element.

Reply:
But temp is reset to soFar on every iteration.

Reply:
that's what we need, preserve the state of `soFar` so that we can append soFar to all the elements `filters[0].ToSearchFilter()`, we don't need "&&" between the elements of `filters[0].ToSearchFilter()`.